hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -339,7 +339,7 @@
       .withDocumentation("");
 
   public static final ConfigProperty<String> EXTERNAL_RECORD_AND_SCHEMA_TRANSFORMATION = ConfigProperty
-      .key(AVRO_SCHEMA + ".externalTransformation")
+      .key(AVRO_SCHEMA.key() + ".externalTransformation")
```

Comment:
when using spark option("hoodie.avro.schema.externalTransformation", true) to pass the config, the `externalTransformation` will be changed to lowercase `externaltransformation` , can we fix it?

Reply:
I see. We can either remove the camel case and break the key as `hoodie.avro.schema.external.transformation`, or just ignore case in comparison. What would you recommend?

Reply:
I prefer `hoodie.avro.schema.external.transformation` to align with other configs.

Reply:
Changed the config key to `hoodie.avro.schema.external.transformation` and also have `hoodie.avro.schema.externalTransformation` as alternative for backwards compatibility.